src/server/serveractiveobject.h

```diff
@@ -241,7 +242,25 @@ class ServerActiveObject : public ActiveObject
 	*/
 	v3s16 m_static_block = v3s16(1337,1337,1337);
 
+	// Names of players to whom the object is to be sent, not considering parents.
+	using Observers = std::optional<std::unordered_set<std::string>>;
+	Observers m_observers;
+
+	// Invalidate final observer cache. This needs to be done whenever
+	// the observers of this object or any of its ancestors may have changed.
+	void invalidateEffectiveObservers();
+	// Get from cache (or compute) final observers.
+	const Observers &getEffectiveObservers();
+	// Force a recalculation of final observers (including all parents).
+	const Observers &recalculateEffectiveObservers();
+	// Check whether an object is finally observed by a player.
+	// Requires the cached final observers to be up to date.
+	bool isEffectivelyObservedBy(const std::string &player_name);
```

Comment:
The object may be observed (managed) or not observed by anyone (unmanaged). The function name and comment is rather misleading for the latter case.
```suggestion
	/// Whether the object may be seen by `player_name`
	bool isVisibleTo(const std::string &player_name);
```

Reply:
This isn't necessarily purely about visuals (though it'll probably be primarily about them), hence why I've been trying to avoid explicitly calling this "visibility" or similar.

I've opted for `isSentTo`, which is effectively what we want to decide (and reads well in the context it's used in, I think). Is that okay?

Reply:
I would interpret `isSentTo` as status query whether the specified player is currently aware of the object. However, the final (distance etc) checks are done in `ActiveObjectMgr`. Your visual "visibility" arguments also make sense ...

Thus it's difficult to determine an accurate name that is lease likely to be misinterpreted. So far I've only come up with `maySendTo`, `mayObserve`, `isObservableBy` as counter-proposals.

Reply:
while a bit unwieldly I think `isEffectivelyObservedBy` is fine.
an alternative is to get rid of the "effective" and rename other variables somehow so the distinction "configured observers" vs. "effective observers" is kept up

Reply:
I have reverted to `isEffectivelyObservedBy` for now: https://github.com/minetest/minetest/pull/13987/commits/14ebd9d9d7406768428f6d973478a4c0bbe36244